pkg/controller/account/iam_test.go

```diff
@@ -256,35 +277,112 @@ func TestDeleteAllAccessKeys(t *testing.T) {
+			setupAWSMock: func(mc *mock.MockClientMockRecorder) {
+				gomock.InOrder(
+					mc.CreateUser(&iam.CreateUserInput{
+						UserName: username,
+					}).Return(
+						&iam.CreateUserOutput{
+							User: &iam.User{
+								UserId:   userID,
+								UserName: username,
+							},
+						},
+						nil, // no error
+					),
+				)
+			},
+			expectedCreateUserOutput: &iam.CreateUserOutput{
+				User: &iam.User{
+					UserId:   userID,
+					UserName: username,
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "InvalidClientTokenId",
+			setupAWSMock: func(mc *mock.MockClientMockRecorder) {
+				gomock.InOrder(
+					mc.CreateUser(&iam.CreateUserInput{
+						UserName: username,
+					}).Return(nil, awserr.New("InvalidClientTokenId", "", nil)).Times(9),
+				)
+			},
+			expectedCreateUserOutput: &iam.CreateUserOutput{},
+			expectedErr:              awserr.New("InvalidClientTokenId", "", nil),
+		},
+		{
+			name: "AccessDenied",
+			setupAWSMock: func(mc *mock.MockClientMockRecorder) {
+				gomock.InOrder(
+					mc.CreateUser(&iam.CreateUserInput{
+						UserName: username,
+					}).Return(nil, awserr.New("AccessDenied", "", nil)).Times(9),
+				)
+			},
+			expectedCreateUserOutput: &iam.CreateUserOutput{},
+			expectedErr:              awserr.New("AccessDenied", "", nil),
+		},
+		{
+			name: "EntityAlreadyExists",
+			setupAWSMock: func(mc *mock.MockClientMockRecorder) {
+				gomock.InOrder(
+					mc.CreateUser(&iam.CreateUserInput{
+						UserName: username,
+					}).Return(nil, awserr.New(iam.ErrCodeEntityAlreadyExistsException, "", nil)),
+				)
+			},
+			expectedCreateUserOutput: &iam.CreateUserOutput{},
+			expectedErr:              awserr.New(iam.ErrCodeEntityAlreadyExistsException, "", nil),
+		},
+		{
+			name: "OtherErr",
+			setupAWSMock: func(mc *mock.MockClientMockRecorder) {
+				gomock.InOrder(
+					mc.CreateUser(&iam.CreateUserInput{
+						UserName: username,
+					}).Return(nil, awserr.New("OtherErr", "", nil)),
+				)
+			},
+			expectedCreateUserOutput: &iam.CreateUserOutput{},
+			expectedErr:              awserr.New("OtherErr", "", nil),
 		},
 	}
-
-	mockAWSClient.EXPECT().CreateUser(&iam.CreateUserInput{
-		UserName: aws.String(usernameStr),
-	}).Return(
-		expectedCreateUserOutput,
-		nil, // no error
-	)
-
-	createUserOutput, err := CreateIAMUser(nullLogger, mockAWSClient, usernameStr)
-	assert.Equal(t, expectedCreateUserOutput, createUserOutput)
-	assert.Nil(t, err)
+	for _, test := range tests {
+		t.Run(
+			test.name,
+			func(t *testing.T) {
+				mocks := setupDefaultMocks(t, []runtime.Object{})
+				test.setupAWSMock(mocks.mockAWSClient.EXPECT())
+				// This is necessary for the mocks to report failures like methods not being called an expected number of times.
+				// after mocks is defined
+				defer mocks.mockCtrl.Finish()
+
+				old := DefaultScale
+				DefaultScale = 0
+				createUserOutput, err := CreateIAMUser(nullLogger, mocks.mockAWSClient, usernameStr)
+				DefaultScale = old
```

Comment:
Similar to `DefaultSleepDelay`, is it local? Could we get away with just re-declaring it once?